packages/downloads/src/hub/hub-request-dataset-export.ts

```diff
@@ -0,0 +1,74 @@
+import { RemoteServerError } from '../remote-server-error';
+import { composeDownloadId } from '../utils';
+import { DownloadFormat } from "../download-format";
+import { convertToHubFormat } from "./format-converter";
+
+export interface IHubDatasetExportRequestParams {
+  host: string;
+  datasetId: string;
+  format: DownloadFormat;
+  spatialRefId?: string;
+  spatialRefWkt?: string;
+  geometry?: string;
+  where?: string;
+}
+
+/**
+ * ```js
+ * import { hubRequestDatasetExport } from "@esri/hub-downloads";
+ * 
+ * const params = {
+ *   host: 'https://hub.arcgis.com,
+ *   datasetId: 'abcdef0123456789abcdef0123456789_0',
+ *   format: 'CSV'
+ * };
+ * hubRequestDatasetExport(params)
+ *   .then(response => {
+ *     // {
+ *     //   downloadId: 'abcdef0123456789abcdef0123456789_0::csv',
+ *     // }
+ *   });
+ * ```
+ * Request an export of a dataset to a particular file format.
+ * @param params - parameters defining a dataset export job
+ */
+export function hubRequestDatasetExport (params: IHubDatasetExportRequestParams) {
+  const {
+    host,
+    datasetId,
+    spatialRefId,
```

Comment:
Since `spatialRefId` is spread out of the params, and then used in `body` it seems like this prop is not actually optional?

Reply:
or is everything downstream from this `null` tolerant?

Reply:
Thanks for catching this.  `spatialRefId` is required for Hub export requests; `geometry` and `where` are optional and the API is tolerant of undefined.  Fixed in c735ae5.